### PR TITLE
New VTK outputs

### PR DESCRIPTION
**Pull request by tkarna**
Updates related to new firedrake VTK output.

### CODE REVIEW

**Comment by tkarna** [comment]
This changes outputs to p1dg for discontinuous fields, p1 otherwise. The visualization spaces and projection/interpolation is still handled in Thetis as Firedrake's `File` object allocates it's own output functions which can be problematic in 3d.

Discontinuous outputs seem to work OK, I'm still seeing a modest increase in memory usage at run time, but it doesn't seem to be associated with the exports. (~6 months ago exporting dg fields used to cause some mysterious memory leak).

File: thetis/exporter.py [hunk 1]
```diff
@@ -5,6 +5,23 @@
...
+    """
+    Figure out the appropriate linear visualization space for fs
+    """
+    is_vector = len(fs.ufl_element().value_shape()) == 1
+    mesh = fs.mesh()
+    family = 'Lagrange' if is_cg(fs) else 'Discontinuous Lagrange'
+    if is_vector:
+        visu_fs = VectorFunctionSpace(mesh, family, 1)
+    else:
+        visu_fs = FunctionSpace(mesh, family, 1)
```
Comment (tkarna) [inline thetis/exporter.py]: Most of the function spaces created here already exist in the Thetis solver object. I presume that function spaces are cached somehow, and this does not actually create any new instances?

Comment (wence-) [inline thetis/exporter.py]: Yes, the heavyweight data is shared between function space instances. So you're paying the cost of a new Python object, but not new dof maps, etc...

Comment (tkarna) [inline thetis/exporter.py]: ok good.

**Comment by tkarna** [comment]
Merging this version to master to fix broken exports.

**Pull request merged**